eng/setup-pr-validation.ps1

```diff
@@ -1,24 +1,40 @@
 [CmdletBinding(PositionalBinding=$false)]
 param (
   [string]$sourceBranchName,
-  [string]$prNumber)
+  [string]$prNumber,
+  [string]$commitSHA)
 
-try {  
-    # name and email are only used for merge commit, it doesn't really matter what we put in there.    
+try {
+    # name and email are only used for merge commit, it doesn't really matter what we put in there.
     git config user.name "RoslynValidation"
-    git config user.email "validation@roslyn.net"   
-    
+    git config user.email "validation@roslyn.net"
+
     if ($sourceBranchName -notlike '*-vs-deps') {
       Write-Host  "##vso[task.LogIssue type=warning;]The base branch for insertion validation is $sourceBranchName, which is not a vs-deps branch."
     }
+
+    Write-Host "Validating the PR head matches the specified commit SHA ($commitSHA)..."
+    if ($commitSHA.Length -lt 7) {
+      Write-Host "##vso[task.LogIssue type=error;]The PR Commit SHA must be at least 7 characters long."
+      exit 1
+    }
+
+    Write-Host "Getting the hash of refs/pull/$prNumber/head..."
+    $remoteRef = git ls-remote origin refs/pull/$prNumber/head
+    Write-Host ($remoteRef | Out-String)
+
+    $prHeadSHA = $remoteRef.Split()[0]
+    if (!$prHeadSHA.StartsWith($commitSHA)) {
```

Comment:
I think it might be reasonable to simply check if the `$commitSHA` is an ancestor of the `$prHeadSHA` because there may be a scenario where the user wants to validate one of the earlier commits in their PR.

Reply:
Yeah, I thought over this, but wasn't sure what the scenario would be. We obviously couldn't take half a PR. Also, It isn't the same as pulling pull/##/merge, since it wouldn't contain changes from the target branch.

Happy to defer this as a future enhancement though, once we work out how it should work.

Reply:
The scenario is that my PR has an RPS regression and I suspect a specific commit in the PR introduced the regression. So I may want to validate the specific commit as well as the one before it. That said, there is no need to introduce this capability, since it would be new.